installation-bundle/src/Resources/views/_configuration_errors.html.twig

```diff
@@ -0,0 +1,78 @@
+{% trans_default_domain 'ContaoInstallationBundle' %}
+{% if errorCode == 1 %}
+    <p class="tl_error">{{ 'unsupported_version'|trans }}</p>
+    <p>{{ 'unsupported_version_explain'|trans|format(version) }}</p>
+{% elseif errorCode == 2 %}
+    <p class="tl_error">{{ 'unsupported_collation'|trans }}</p>
+    <p>{{ 'unsupported_collation_explain'|trans|format(collation)|raw }}</p>
+    <div id="sql_wrapper">
+    <pre>doctrine:
+    dbal:
+        connections:
+            default:
+                default_table_options:
+                    charset: utf8
+                    collate: utf8_unicode_ci</pre>
+    </div>
+{% elseif errorCode == 3 %}
+    <p class="tl_error">{{ 'unsupported_engine'|trans }}</p>
+    <p>{{ 'unsupported_engine_explain'|trans|format(engine)|raw }}</p>
+    <div id="sql_wrapper">
+    <pre>doctrine:
+    dbal:
+        connections:
+            default:
+                default_table_options:
+                    engine: MyISAM
+                    row_format: ~</pre>
+    </div>
+{% elseif errorCode == 4 %}
+    <p class="tl_error">{{ 'engine_mismatch'|trans }}</p>
+    <p>{{ 'engine_mismatch_explain'|trans|format(engine)|raw }}</p>
+    <div id="sql_wrapper">
+    <pre>doctrine:
+    dbal:
+        connections:
+            default:
+                default_table_options:
+                    charset: utf8
+                    collate: utf8_unicode_ci</pre>
+    </div>
+{% elseif errorCode == 5 %}
+    <p class="tl_error">{{ 'innodb_large_prefix'|trans }}</p>
+    <p>{{ 'innodb_large_prefix_explain'|trans|raw }}</p>
+    <div id="sql_wrapper">
+    <pre>doctrine:
+    dbal:
+        connections:
+            default:
+                default_table_options:
+                    charset: utf8
+                    collate: utf8_unicode_ci</pre>
+    </div>
+{% elseif errorCode == 6 %}
+    <p class="tl_error">{{ 'innodb_additional'|trans }}</p>
+    <p>{{ 'innodb_additional_explain'|trans|raw }}</p>
+    <div id="sql_wrapper">
+    <pre>innodb_large_prefix = 1
+innodb_file_format = Barracuda
+innodb_file_per_table = 1</pre>
+    </div>
+{% elseif errorCode == 7 %}
+    <p class="tl_error">{{ 'strict_sql_mode'|trans }}</p>
+    <p>{{ 'strict_sql_mode_explain'|trans|raw }}</p>
+    <div id="sql_wrapper">
+    <pre>doctrine:
+  dbal:
+    connections:
+      default:
+        options:
+        {%- if driver == 'mysqli' %}
+          # Define a MYSQLI_INIT_COMMAND
+          3: "SET SESSION sql_mode=(SELECT CONCAT(@@sql_mode, ',TRADITIONAL'))"
+        {%- else %}
+          # Define a PDO::MYSQL_ATTR_INIT_COMMAND
+          1002: "SET SESSION sql_mode=(SELECT CONCAT(@@sql_mode, ',TRADITIONAL'))"
```

Comment:
We might want to swap out `TRADITIONAL` with `STRICT_TRANS_TABLES` to be less strict (and cause less friction with extensions and the like).

See MySQL docs:
> Before MySQL 5.7.4, and in MySQL 5.7.8 and later, TRADITIONAL is equivalent to STRICT_TRANS_TABLES, STRICT_ALL_TABLES, NO_ZERO_IN_DATE, NO_ZERO_DATE, ERROR_FOR_DIVISION_BY_ZERO, NO_AUTO_CREATE_USER, and NO_ENGINE_SUBSTITUTION.

> From MySQL 5.7.4 though 5.7.7, TRADITIONAL is equivalent to STRICT_TRANS_TABLES, STRICT_ALL_TABLES, NO_AUTO_CREATE_USER, and NO_ENGINE_SUBSTITUTION. The NO_ZERO_IN_DATE, NO_ZERO_DATE, and ERROR_FOR_DIVISION_BY_ZERO modes are not named because in those versions their effects are included in the effects of strict SQL mode (STRICT_ALL_TABLES or STRICT_TRANS_TABLES). Thus, the effects of TRADITIONAL are the same in all MySQL 5.7 versions (and the same as in MySQL 5.6). For additional discussion, see SQL Mode Changes in MySQL 5.7.

Reply:
I’d vote for keeping `TRADITIONAL` (as the suggestion), that would make the behavior across installations very equal. And we are only requiring `STRICT_TRANS_TABLES` to not show the warning, right?